pkg/util/push/push_test.go

```diff
@@ -164,6 +177,33 @@ func createRequest(t testing.TB, protobuf []byte) *http.Request {
 	return req
 }
 
+func createOTLPRequest(t testing.TB, protoMetrics *metricpb.ResourceMetrics) *http.Request {
+	t.Helper()
+
+	pbRequest := &colmetricpb.ExportMetricsServiceRequest{
+		ResourceMetrics: []*metricpb.ResourceMetrics{protoMetrics},
+	}
+
+	rawBytes, err := golangproto.Marshal(pbRequest)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("POST", "http://localhost/", bytes.NewReader(rawBytes))
+	require.NoError(t, err)
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	return req
+}
+
+func createOTLPProtobuf(t testing.TB) *metricpb.ResourceMetrics {
+	return otlpMetricsFromPRWBytes(t, createPrometheusRemoteWriteProtobuf(t))
+}
+
+func otlpMetricsFromPRWBytes(t testing.TB, input []byte) *metricpb.ResourceMetrics {
+	prwReq := &prompb.WriteRequest{}
+	require.NoError(t, proto.Unmarshal(input, prwReq))
+
+	return e2emimir.TimeseriesToOTLPResourceMetrics(prwReq.Timeseries)
```

Comment:
Nit: not sure if there's already code like this in the repo, but I'd move `TimeseriesToOTLPResourceMetrics` here to `util` and make `e2emimir` depend on `pkg/util` instead of having `pkg/util` depending on `e2emimir`.